File: service/src/main/java/hu/bsstudio/robonaut/team/query/QueryTeamData.java
package hu.bsstudio.robonaut.team.query;

import hu.bsstudio.robonaut.team.TeamService;
import hu.bsstudio.robonaut.team.query.model.Requester;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class QueryTeamData {

    @Autowired
    private TeamService teamService;

    @RabbitListener(queues = "general.teamData")
    public void sendTeamData(final Requester requester) {
        LOG.info("Teams were requested by {}", requester);
        teamService.findAllTeam()
            .subscribe();
    }
}

File: service/src/main/kotlin/hu/bsstudio/robonaut/team/query/QueryTeamData.kt
package hu.bsstudio.robonaut.team.query

import hu.bsstudio.robonaut.team.TeamService
import org.slf4j.LoggerFactory
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Component

@Component
class QueryTeamData(private val teamService: TeamService) {
    @RabbitListener(queues = ["general.teamData"])
    fun sendTeamData() {
        LOG.info("Teams were requested.")
        teamService.findAllTeam()
            .subscribe()
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(this::class.java)
    }
}
